app/thermal_management/thermalmgmt.c

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <logging/log.h>
+#include "thermalmgmt.h"
+#include "fan.h"
+#include "thermal_sensor.h"
+#include "board_config.h"
+#include "smc.h"
+#include "peci_hub.h"
+#include "pwrplane.h"
+
+LOG_MODULE_REGISTER(thermal, CONFIG_THERMAL_MGMT_LOG_LEVEL);
+
+/* PECI takes some time to initialize in CPU side, initial request will fail
+ * after reaching S0.
+ * Hence allow 1 sec time before reading temperature from peci.
+ * Till then report fail safe temperature of 28C.
+ */
+#define CPU_TEMP_ACCESS_DELAY_SEC		2U
+
+#define CPU_FAIL_SAFE_TEMPERATURE		28U
+
+/* CPU fail critical temperature value is 72C */
+#define CPU_FAIL_CRITICAL_TEMPERATURE		72U
+
+/*
+ * EC Self control fan speed based on CPU temperature info.
+ *
+ * Two simple approaches are possible
+ * Approach #1 Control fan with liner profile.
+ * For CPU temperature between 0 to 100, fan to also rotate at equivalent speed,
+ * but rather than changing speed on every single degree change, it is changed
+ * after 8 degrees.
+ *
+ * Approach #2 Use a lookup table based on temperature
+ */
+#define GET_FAN_SPEED_FOR_TEMP(temp)		(temp & (~0x7))
+
+struct therm_sensor *therm_sensor_tbl;
+struct fan_dev *fan_dev_tbl;
+static u8_t max_adc_sensors;
+static u8_t max_fan_dev;
+static bool thermal_initialized;
+static bool peci_initialized;
+static u8_t fan_duty_cycle[FAN_DEV_TOTAL];
+static bool fan_duty_cycle_change;
+static int cpu_temp;
+/* Using module property, before introducing straps handler */
+static bool fan_override;
+
+static uint8_t mapped_fan_speed(void)
```

Comment:
if rest other code is all using u8_t, do you want to change all occurrences of uint8_t to u8_t as well?

Reply:
Yes, that should be the right thing to do in this case